yolox/utils/mlflow_logger.py

```diff
@@ -0,0 +1,373 @@
+                self._ml_flow.start_run(run_name=self.run_name, nested=self._nested_run)
+                logger.debug(
+                    f"MLflow run started with run_id={self._ml_flow.active_run().info.run_id}")
+                self._auto_end_run = True
+                self._initialized = True
+            # filters these params from args
+            keys = ['experiment_name', 'batch_size', 'exp_file', 'resume', 'ckpt', 'start_epoch',
+                    'num_machines', 'fp16', 'logger']
+            combined_dict = {k: v for k, v in vars(args).items() if k in keys}
+            if exp is not None:
+                exp_dict = self.convert_exp_todict(exp)
+                combined_dict = {**exp_dict, **combined_dict}
+            self.log_params_mlflow(combined_dict)
+            mlflow_tags = os.getenv("MLFLOW_TAGS", None)
+            if mlflow_tags:
+                mlflow_tags = json.loads(mlflow_tags)
+                self._ml_flow.set_tags(mlflow_tags)
+
+    def log_params_mlflow(self, params_dict):
+        """
+        Log hyperparameters to MLflow.
+        MLflow's log_param() only accepts values no longer than 250 characters.
+        No overwriting of existing parameters is allowed by default from mlflow.
+
+        Input:
+            params_dict: dict of hyperparameters
+
+        """
+
+        if is_main_process():
+            params_dict = self.flatten_dict(params_dict) if self._flatten_params else params_dict
+            # remove params that are too long for MLflow
+            for name, value in list(params_dict.items()):
+                # internally, all values are converted to str in MLflow
+                if len(str(value)) > self._MAX_PARAM_VAL_LENGTH:
+                    logger.warning(
+                        f'Trainer is attempting to log a value of "{value}" for key "{name}" as a '
+                        f'parameter. MLflow\'s log_param() only accepts values no longer than 250 '
+                        f'characters so we dropped this attribute. You can use '
+                        f'`MLFLOW_FLATTEN_PARAMS` environment variable to flatten the parameters '
+                        f'and avoid this message.'
+                    )
+                    del params_dict[name]
+            # MLflow cannot log more than 100 values in one go, so we have to split it
+            combined_dict_items = list(params_dict.items())
+            for i in range(0, len(combined_dict_items), self._MAX_PARAMS_TAGS_PER_BATCH):
+                if self._async_log:
+                    self._ml_flow.log_params(
+                        dict(combined_dict_items[i: i + self._MAX_PARAMS_TAGS_PER_BATCH]),
+                        synchronous=False
+                    )
+                else:
+                    self._ml_flow.log_params(
+                        dict(combined_dict_items[i: i + self._MAX_PARAMS_TAGS_PER_BATCH])
+                    )
+
+    def convert_exp_todict(self, exp):
+        """
```

Comment:
This project use google style docstring.
For example
```
decription of function/method

Args:
     arg (type): description

Returns:
```
Please update your docstring in a unified format. Thanks : )